src/Akka.Persistence.SqlServer/Journal/QueryBuilder.cs

```diff
@@ -126,29 +127,11 @@ public DbCommand InsertBatchMessages(IPersistentRepresentation[] messages)
             return command;
         }
 
-        public DbCommand DeleteBatchMessages(string persistenceId, long toSequenceNr, bool permanent)
-        {
-            var sql = BuildDeleteSql(toSequenceNr, permanent);
-            var command = new SqlCommand(sql)
-            {
-                Parameters = { PersistenceIdToSqlParam(persistenceId) }
-            };
-
-            return command;
-        }
-
-        private string BuildDeleteSql(long toSequenceNr, bool permanent)
+        public DbCommand DeleteBatchMessages(string persistenceId, long toSequenceNr)
         {
             var sqlBuilder = new StringBuilder();
 
-            if (permanent)
-            {
-                sqlBuilder.Append("DELETE FROM {0}.{1} ".QuoteSchemaAndTable(_schemaName, _tableName));
-            }
-            else
-            {
-                sqlBuilder.Append("UPDATE {0}.{1} SET IsDeleted = 1 ".QuoteSchemaAndTable(_schemaName, _tableName));
-            }
+            sqlBuilder.Append("DELETE FROM {0}.{1} ".QuoteSchemaAndTable(_schemaName, _tableName));
```

Comment:
I think, this can be cached now in some sort of field. No need to use `StringBuilder` + string format each time.